apps/marketplace/actions/supplierActions.js

```diff
@@ -0,0 +1,16 @@
+import dmapi from '../services/apiClient'
+
+const findSuppliers = keyword => {
+  const params = {
+    keyword
+  }
+  return dmapi({ url: `/suppliers/search`, params }).then(response => {
+    if (!response || response.error) {
+      throw response.errorMessage
```

Comment:
The use of `throw` here is a bit different to how we've previously been using `dmapi()`. A goal of this component was to make it without a dependency on React Redux, so the whole `dispatch()` system is unavailable. It turns out that `dmapi()` and its response object is a `Promise`, which means handling success and fail can be done without callbacks - this is pointed out in the component code below.